lib/http2/hpack.c

```diff
@@ -213,8 +213,10 @@ static h2o_iovec_t *decode_string(h2o_mem_pool_t *pool, const uint8_t **src, con
         if (len > src_end - *src)
             return NULL;
         ret = alloc_buf(pool, len * 2); /* max compression ratio is >= 0.5 */
-        if ((ret->len = h2o_hpack_decode_huffman(ret->base, *src, len, is_header_name, err_desc)) == SIZE_MAX)
+        if ((ret->len = h2o_hpack_decode_huffman(ret->base, *src, len, is_header_name, err_desc)) == SIZE_MAX) {
+            h2o_mem_release_shared(ret);
```

Comment:
Fuzzer did not like this change. Looks like I misunderstood how the pool works... if allocated from a pool we should never free it manually, right? I think I'll need to drop this patch.

Reply:
Right. Consider memory pool as a pool that you can allocate memory from, but never release. All the memory associated to the pool is released when the pool is destructed.